### PR TITLE
Cleanup: use new_resource everywhere

### PR DESCRIPTION
**Pull request by olive42**
Consistency: use api_endpoint and api_username in resources/repo.rb, and use
default nexus3 api attributes for API credentials.

### CODE REVIEW

File: attributes/api.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@
 default['nexus3']['api']['type'] = 'groovy'
 default['nexus3']['api']['host'] = 'http://localhost:8081'
 default['nexus3']['api']['endpoint'] = "#{node['nexus3']['api']['host']}/service/siesta/rest/v1/script/"
+default['nexus3']['api']['username'] = 'admin'.freeze
```
Comment (jmauro) [inline attributes/api.rb]: No need freeze attribute

File: resources/repo.rb [hunk 2]
```diff
@@ -2,17 +2,18 @@
 property :repo_type, String, desired_state: false, identity: true, default: 'maven2-hosted'.freeze
 property :attributes, Hash, default: lazy { ::Mash.new } # Not mandatory but strongly recommended in the generic case.
 property :online, [true, false], default: true
-property :api_url, String, desired_state: false, identity: true, default: 'http://localhost:8081/service/siesta/rest/v1/script/'.freeze
-property :api_user, String, desired_state: false, identity: true, default: 'admin'.freeze
-property :api_password, String, desired_state: false, identity: true, sensitive: true, default: 'admin123'.freeze
+property :api_endpoint, String, desired_state: false, identity: true, default: node['nexus3']['api']['endpoint']
```
Comment (jmauro) [inline resources/repo.rb]: lazy attribute

File: resources/repo.rb [hunk 3]
```diff
@@ -2,17 +2,18 @@
...
+property :api_username, String, desired_state: false, identity: true, default: node['nexus3']['api']['username']
```
Comment (jmauro) [inline resources/repo.rb]: lazy attribute

File: resources/repo.rb [hunk 4]
```diff
@@ -2,17 +2,18 @@
...
+property :api_password, String, desired_state: false, identity: true, sensitive: true,
+                                default: node['nexus3']['api']['password']
```
Comment (jmauro) [inline resources/repo.rb]: lazy attributes

File: resources/repo.rb [hunk 5]
```diff
@@ -2,17 +2,18 @@
...
 
 def apiclient
-  @apiclient ||= ::Nexus3::Api.new(api_url, api_user, api_password)
+  @apiclient ||= ::Nexus3::Api.new(api_endpoint, api_username, api_password)
```
Comment (jmauro) [inline resources/repo.rb]: new_resource must be put in place

File: test/fixtures/cookbooks/nexus3_test/attributes/default.rb [hunk 6]
```diff
@@ -1,2 +1,3 @@
-default['nexus3_test']['connection_retries'] = 0
-default['nexus3']['version'] = '3.3.1-01'
+default['nexus3']['version'] = '3.3.1-01'.freeze
```
Comment (jmauro) [inline test/fixtures/cookbooks/nexus3_test/attributes/default.rb]: remove freeze this an attributes

File: attributes/api.rb (hunk 1 above)
Comment (olive42) [reply attributes/api.rb]: done

File: resources/repo.rb (hunk 2 above)
Comment (olive42) [reply resources/repo.rb]: done

File: resources/repo.rb (hunk 3 above)
Comment (olive42) [reply resources/repo.rb]: done

File: resources/repo.rb (hunk 4 above)
Comment (olive42) [reply resources/repo.rb]: done

File: resources/repo.rb (hunk 5 above)
Comment (olive42) [reply resources/repo.rb]: done and moved apiclient method inside load_current_value

File: test/fixtures/cookbooks/nexus3_test/attributes/default.rb (hunk 6 above)
Comment (olive42) [reply test/fixtures/cookbooks/nexus3_test/attributes/default.rb]: done

**Pull request merged**